### PR TITLE
feat: Data Table: Searchable and Unsearchable columns

### LINKED ISSUE #513: Data Table: Searchable and unsearchable columns.
Is it possible to flag a column to not to be searchable?
If yes, how do I do that.

### PR DESCRIPTION
**Pull request by jeremysmartt**
## Description

Added notsearchable property on columns to keep certain columns from being searched on in the filterData method. Also added unit tests for table and updated documentation.

Closes #513

### What's included?

- modified:   src/app/components/components/data-table/data-table.component.html
- modified:   src/app/components/components/data-table/data-table.component.ts
- modified:   src/platform/core/data-table/data-table.component.ts
- modified:   src/platform/core/data-table/services/data-table.service.ts
- new file:   src/platform/core/data-table/data-table.component.spec.ts

#### Test Steps

- [x] checkout branch
- [x] Go to http://localhost:4200/#/components/data-table
- [x] Go to the "Data Table with components"
- [x] Click the Search magnify glass
- [x] type in 392 see no results
- [x] delete 392 from search box
- [x] Click the TOGGLE SEARCHING CALORIES COLUMN button
- [x] See 1 result row in the table
- [x] Make sure documentation makes sense

#### General Tests for Every PR

- [ ] `ng serve --aot` still works.
- [x] `npm run lint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build` still works.

##### Screenshots or link to CodePen/Plunker/JSfiddle
non-searchable-columns

### CODE REVIEW

**Comment by kyleledbetter** [comment]
love the PR, could we name the input something more simple like `filter="false"` and true is the default?

**Comment by emoralesb05** [comment]
I feel that this is more a design pattern than an enhancement.. not sure about this yet.

**Comment by jeremysmartt** [comment]
Right at first I thought maybe a design pattern, but then actually I don't think it is.  I think the ability to pass in the columns and get hits based on properties will continue to be needed.  For example, probably want to do the same thing when a column is hidden:
https://github.com/Teradata/covalent/issues/511

**Comment by emoralesb05** [comment]
The use case is legit, just debating if it should be part of the interface or applied in a different way since this seems more of an sorting implementation thing.

In both cases, there is stuff to be done in `TdDataTableService` for sure since thats a sorting implementation.

**Comment by jeremysmartt** [comment]
@kyleledbetter I updated the name of the input to be filter="false" and true is the default.

**Comment by kyleledbetter** [comment]
since we have a new batch of filtering/hiding/showing features coming, thinking we can do a new demo since that current demo is getting a little convoluted

**Comment by kyleledbetter** [comment]
maybe not another demo, but maybe update the current complex datatable demo to use slide toggles for options instead of md-buttons, i'll take a stab

**Comment by thomaslennan** [comment]
Another framework provides the ability to define both global and individual filters. Wondering if this PR should really be addressed by being able to filter on a per column basis. I, as a user, would find it confusing that I saw data in a column of the table that wasn't visible after input of some global filter value that matched the value that saw in that column. Alternatively, I'd want some visual indicator that the particular column isn't participating in the filter?

File: src/platform/core/data-table/services/data-table.service.ts [hunk 1]
```diff
@@ -13,14 +13,16 @@ export class TdDataTableService {
    *
    * Searches [data] parameter for [searchTerm] matches and returns a new array with them.
    */
-  filterData(data: any[], searchTerm: string, ignoreCase: boolean = false): any[] {
+  filterData(data: any[], searchTerm: string, ignoreCase: boolean = false, nonSearchAbleColumns?: string[]): any[] {
```
Comment (emoralesb05) [inline src/platform/core/data-table/services/data-table.service.ts]: Should we just call this `excludedColumns`?

File: src/platform/core/data-table/data-table.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import 'hammerjs';
+import { Component } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { TdDataTableComponent, ITdDataTableColumn } from './data-table.component';
+import { TdDataTableService } from './services/data-table.service';
+import { CovalentDataTableModule } from './data-table.module';
+import { NgModule, DebugElement } from '@angular/core';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+describe('Component: TdPagingBarComponent', () => {
```
Comment (emoralesb05) [inline src/platform/core/data-table/data-table.component.spec.ts]: should be `'Component: DataTable'`.

File: src/app/components/components/data-table/data-table.component.html [hunk 3]
```diff
@@ -328,7 +328,7 @@ <h4 class="md-subhead">Paging Bar / Search Box / Sortable components</h4>
 
               filter(): void {
                 let newData: any[] = this.data;
-                newData = this._dataTableService.filterData(newData, this.searchTerm, true);
+                newData = this._dataTableService.filterData(newData, this.searchTerm, true, this.columns);
```
Comment (emoralesb05) [inline src/app/components/components/data-table/data-table.component.html]: Demo needs to be updated with latest `API` changes.

File: src/platform/core/data-table/services/data-table.service.ts (hunk 1 above)
Comment (jeremysmartt) [inline src/platform/core/data-table/services/data-table.service.ts]: updated

File: src/platform/core/data-table/data-table.component.spec.ts (hunk 2 above)
Comment (jeremysmartt) [inline src/platform/core/data-table/data-table.component.spec.ts]: updated

File: src/app/components/components/data-table/data-table.component.html (hunk 3 above)
Comment (jeremysmartt) [inline src/app/components/components/data-table/data-table.component.html]: updated

**Pull request merged**